app/src/main/java/de/tum/in/tumcampusapp/service/WifiScanHandler.kt

```diff
@@ -35,8 +36,9 @@ class WifiScanHandler : Handler() {
     }
 
     fun onScanFinished() {
-        if (wifiLock.isHeld) {
-            wifiLock.release()
+        wifiLock?.apply {
```

Comment:
Please use `let` instead of `apply`, as this is the general practice for null checks. 

`apply` is typically used for scenarios where you will return the variable, e.g. `val x = SomeObject().apply { doSomethingOnSomeObject() }`